Server/cron/cronFunctions/qualifierMatchup.ts

```diff
@@ -22,30 +27,46 @@ async function initialize (): Promise<CronJobData[]> {
         cronJobs = cronJobs.filter(j => j.date.getTime() > Date.now());
         cronJobs.push({
             type: CronJobType.QualifierMatchup,
-            date: new Date(Date.now() + 60 * 1000), // 1 minute delay to avoid Date in past error
+            date: new Date(Date.now() + 10 * 1000), // 10 second delay to avoid Date in past error
         });
     }
 
     return cronJobs;
 }
 
 async function execute (job: CronJobData) {
-    const futureDate = new Date();
-    futureDate.setMinutes(futureDate.getMinutes() + 15);
+    const futureDate = new Date(job.date.getTime() + 15 * 60 * 1000);
 
     // Get all matchups that are in the past and have not been played
     const matchups = await Matchup
         .createQueryBuilder("matchup")
+        .leftJoinAndSelect("matchup.referee", "referee")
+        .leftJoinAndSelect("matchup.streamer", "streamer")
         .innerJoinAndSelect("matchup.stage", "stage")
+        .innerJoinAndSelect("stage.mappool", "mappool")
+        .innerJoinAndSelect("mappool.slots", "slot")
+        .innerJoinAndSelect("slot.maps", "map")
+        .innerJoinAndSelect("map.beatmap", "beatmap")
         .innerJoinAndSelect("stage.tournament", "tournament")
+        .innerJoinAndSelect("tournament.organizer", "organizer")
         .leftJoinAndSelect("matchup.teams", "team")
         .leftJoinAndSelect("team.manager", "manager")
         .leftJoinAndSelect("team.members", "member")
         .where("matchup.date <= :now", { now: futureDate })
+        .andWhere("stage.stageType = 1")
         .andWhere("matchup.mp IS NULL")
         .getMany();
 
-    // TODO: For each matchup, create and run a multiplayer match
+    matchups.forEach(matchup => {
+        runMatchup(matchup).catch(err => {
+            if (err) {
+                console.log(err);
+                const channel = discordClient.channels.cache.get(config.discord.coreChannel);
+                if (channel instanceof TextChannel)
+                    channel.send(`Error running match GHIVE THIS IMMEDIATE ATTENTION:\n\`\`\`\n${err}\n\`\`\``);
+            }
+        });
+    });
```

Comment:
nope. this will be executed on cron runner rather than the bancho bot. i guess you can have the bancho bot also handling cron, but then please whitelist it to running matches tasks (and blacklist that kind of task on the general cron runner)

Reply:
Yea it was running in the cron-runner instance, idk how to signal banchobot to run it after cron timer is finished or how to whitelist it to run matchup tasks

Reply:
so how was it connected to bancho in your testing? does every instance of this project connect to bancho? if yes, we'll hit connections limit pretty fast. only the bancho bot entrypoint should connect to bancho

Reply:
yea it was also running in api-dev too Oops, i fixed both comments in https://github.com/Corsace/Corsace/pull/213/commits/f63af983ee3787a47d70c9da6d2d31de56357851